src/components/select_device_popup.rs

```diff
@@ -59,6 +59,22 @@ impl SelectDevicePopupComponent {
             .send(Action::HideSelectDevicePopUp.into())?;
         Ok(())
     }
+
+    fn show_select_flavor(&self) -> Result<()> {
+        self.action_tx
+            .as_ref()
+            .ok_or_else(|| eyre!("action_tx is None"))?
+            .send(Action::ShowSelectFlavorPopUp.into())?;
+        Ok(())
+    }
+
+    fn load_flavors(&self) -> Result<()> {
```

Comment:
Please load flavors as soon as loading supported platform finished. You don't need to wait until user actually click `f` key.